### PR TITLE
Extract MAP_HELPER module.

### PR DESCRIPTION
**Pull request by di-wu**
- Extracted the MAP_HELPER module to a separated file.
- Added test and documentation to the module.
- Added CI for tests.

### CODE REVIEW

File: src/mapHelper.mo [hunk 1]
```diff
@@ -0,0 +1,111 @@
+import Array "mo:base/Array";
+import HashMap "mo:base/HashMap";
+
+// MapHelper contains helper function to interact with hash maps.
+module {
```
Comment (SuddenlyHazel) [inline src/mapHelper.mo:5]: Wonderfully verbose! 🎉

File: test/mapHelper.mo [hunk 2]
```diff
@@ -0,0 +1,54 @@
+import HashMap "mo:base/HashMap";
+import Principal "mo:base/Principal";
+import Debug "mo:base/Debug";
+
+import H "../src/mapHelper";
+
```
Comment (SuddenlyHazel) [inline test/mapHelper.mo:6]: Thanks for adding these

File: src/mapHelper.mo [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+                    case (? v) {};
+                };
+            };
+        };
+        return true;
+    };
+
+    // Filters out all elements based on the given seach function `f`.
+    // If no elements match, the entry is removed from the hash map.
+    public func filter<K, V>(
```
Comment (SuddenlyHazel) [inline src/mapHelper.mo:84]: This naming makes a ton more sense 👍

**Review by SuddenlyHazel — APPROVED** [review]

File: src/mapHelper.mo (hunk 3 above)
Comment (di-wu) [reply src/mapHelper.mo:84]: Yeah, had some difficulties understanding `remove` at first.

**Pull request merged**